pkg/clusteragent/admission/mutate/autoinstrumentation/injector.go

```diff
@@ -114,6 +116,36 @@ func (i *injector) initContainer() initContainer {
 }
 
 func (i *injector) requirements() libRequirement {
+	envVars := []envVar{
+		{
+			key:     "LD_PRELOAD",
+			valFunc: joinValFunc(asAbs(injectorFilePath("launcher.preload.so")), ":"),
+		},
+		{
+			key:     "DD_INJECT_SENDER_TYPE",
+			valFunc: identityValFunc("k8s"),
+		},
+		{
+			key:     "DD_INJECT_START_TIME",
+			valFunc: identityValFunc(strconv.FormatInt(i.injectTime.Unix(), 10)),
+		},
+	}
+	if i.debug {
+		envVars = append(envVars, []envVar{
```

Comment:
You don't need to do an intermediate slice and `...` here, you could do:

```go
trueVal := identityValFunc("true")
envVars = append(
    envVars, 
    { key: "DD_APM_INSTRUMENTATION_DEBUG", valFunc: trueVal },
    { key: "DD_TRACE_STARTUP_LOGS", valFunc: trueVal },
    { key: "DD_TRACE_DEBUG", valFunc: trueVal },
)
```

Reply:
The other option is to declare the `debugEnvVars` outside of this function and to then  have a helper do:

```go
var debugEnvVars = []envVar{...}
func (i *injector) debugEnvVars() []envVar {
    if i.debug {
        return debugEnvVars
    }
    
    return nil
}
```

And in `requirements()` this would work: 

```go
envVars: append($EXISTING_CODE, i.debugEnvVars()...),
```